types/ember/test/ember-module-tests.ts

```diff
@@ -0,0 +1,280 @@
+// on
+const o3 = EmberObject.extend({
+    name: 'foo',
+    nameWatcher: Ember.on('init', () => {}),
+    nameWatcher2: Ember.on('destroy', () => {})
+});
+// removeListener
+Ember.addListener(o2, 'create', () => {});
+Ember.addListener({}, 'create', () => {}); // $ExpectError
+// removeObserver
+Ember.removeObserver(o2, 'create', () => {});
+Ember.removeObserver({}, 'create', () => {}); // $ExpectError
+// runInDebug
+Ember.runInDebug(() => {});
+// sendEvent
+Ember.sendEvent(o2, 'clicked', [1, 2]);
+// set
+Ember.set(o2.create(), 'name', 'bar'); // $ExpectType string
+Ember.set(o2.create(), 'nam', 'bar'); // $ExpectError
+// setOwner
+Ember.setOwner(o2.create(), {});
+// setProperties
+Ember.setProperties(o2.create(), { name: 'bar' }).name; // $ExpectType string
+// tryInvoke
+Ember.tryInvoke(o2, 'init');
+Ember.tryInvoke(o2, 'init', [441]);
+// trySet
+Ember.trySet(o2, 'nam', ''); // $ExpectType any
+// typeOf
+Ember.typeOf(''); // $ExpectType "string"
+Ember.typeOf(Ember.A()); // $ExpectType "array"
+// warn
+Ember.warn('be caseful!');
+// VERSION
+Ember.VERSION; // $ExpectType string
+
+// onerror
+
+Ember.onerror = (err: Error) => console.error(err);
+Ember.onerror = (num: number, err: Error) => console.error(err); // $ExpectError
+
+// Classes
+// TODO ContainerProxyMixin
+// Ember
+// Ember.Application
+new Ember.Application(); // $ExpectType Application
+Ember.Application.create(); // $ExpectType Application
+// Ember.ApplicationInstance
+new Ember.ApplicationInstance(); // $ExpectType ApplicationInstance
+Ember.ApplicationInstance.create(); // $ExpectType ApplicationInstance
+// TODO: Ember.ApplicationInstance.BootOptions
+// Ember.Array
+const a1: Ember.Array<string> = [];
+// Ember.ArrayProxy
+new Ember.ArrayProxy<number>([3, 3, 2]); // $ExpectType ArrayProxy<number>
+// Ember.Checkbox
+const cb = new Ember.Checkbox(); // $ExpectType Checkbox
+cb.tagName; // $ExpectType string
+// Ember.Component
+const C1 = Ember.Component.extend({ classNames: ['foo'] });
+class C2 extends Ember.Component {
+    classNames = ['foo'];
+}
+const c1 = new C1();
+const c2 = new C2();
+C1.create();
+C2.create();
+c1.didInsertElement();
+c2.didInsertElement();
+// Ember.ComputedProperty
+const cp: Ember.ComputedProperty<string, string> = Ember.computed('foo', {
+    get(): string {
+        return '';
+    },
+    set(_key: string, newVal: string): string {
+        return '';
+    }
+});
+// Ember.ContainerDebugAdapter
+const cda = new Ember.ContainerDebugAdapter(); // $ExpectType ContainerDebugAdapter
+// Ember.Controller
+const con = new Ember.Controller(); // $ExpectType Controller
+// Ember.CoreObject
+const co = new Ember.CoreObject(); // $ExpectType CoreObject
+// Ember.DataAdapter
+const da = new Ember.DataAdapter(); // $ExpectType DataAdapter
+// Ember.Debug
+Ember.Debug.registerDeprecationHandler(() => {});
+Ember.Debug.registerWarnHandler(() => {});
+// Ember.DefaultResolver
+const dr = new Ember.DefaultResolver();
+dr.resolve('route:index');
```

Comment:
needs ExpectType {}

Reply:
test for error case?

Reply:
Added the error case. Leaving the `ExpectType` for when we improve registry/container type-checking. A `{}` isn't even the correct type for this function to be returning.